backend/src/main/java/com/aadeshandreas/ailearning/ai_learning_companion/controller/DocumentController.java

```diff
@@ -54,6 +55,30 @@ public ResponseEntity<ApiResponse<?>> uploadDocument(@RequestParam("file") Multi
         }
     }
 
```

Comment:
```suggestion
/**
 * Handles the HTTP POST request to upload raw text content for processing.
 * The uploaded text is wrapped in a {@link DocumentText} model and stored
 * in the user's session for further operations, such as generating summaries
 * or flashcards.
 *
 * <p>Validation is performed by the {@link DocumentText} model itself;
 * empty or null text will trigger an {@link IllegalArgumentException}.</p>
 *
 * @param documentText The {@link DocumentText} object containing the text input.
 * @return A {@link ResponseEntity} wrapping a generic {@link ApiResponse}. On success,
 *         it returns a 200 OK status with a success message. If the text is invalid,
 *         it returns a 400 Bad Request. Unexpected errors result in a 500 Internal Server Error.
 */
```

The method should have a JavaDoc.
Here’s a draft you can use if you want :).